video/worker.go

```diff
@@ -0,0 +1,80 @@
+package video
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/lbryio/transcoder/encoder"
+	"github.com/lbryio/transcoder/formats"
+	"github.com/lbryio/transcoder/pkg/timer"
+	"github.com/lbryio/transcoder/queue"
+)
+
+func SpawnProcessing(videoPath string, q *queue.Queue, lib *Library) {
+	p := q.StartPoller()
+	logger.Info("started video processor")
+	for t := range p.IncomingTasks() {
```

Comment:
you have to check if this channel is closed. what if poller was shut down?

Reply:
But loop will terminate if the channel closes.